librecval/extract_phrases.py

```diff
@@ -50,34 +52,47 @@ class MissingMetadataError(RuntimeError):
     """
 
 
-class InvalidTextGridName(RuntimeError):
+class InvalidFileName(RuntimeError):
     """
-    Raised when the TextGrid name does not follow an appropriate pattern.
+    Raised when the ELAN name does not follow an appropriate pattern.
     """
 
 
+class MissingTranslationError(RuntimeError):
+    """
+    Raised when the 'English (word)' and 'English (sentence)' tiers
+    can not be found in a .eaf file suggesting that the word or phrase
+    does not have an existing translation
+    """
+
+    # TODO: this is a weird assumption to make. So far it hasn't been an issue though?
+
+
 # ########################################################################## #
 
 
-class RecordingInfo(NamedTuple):
+class Segment(NamedTuple):
     """
-    All the information you could possible want to know about a recorded
-    snippet.
+    Stores an audio segment extracted from a .eaf file
     """
 
-    session: SessionID
+    translation: str  # in English
+    transcription: str  # in Cree
+    type: str  # "word" or "sentence"
```

Comment:
I like this suggestion, but I'm hesitant to incorporate something that might not work. I do think this is clearer, but when the extraction process takes 4+ hours to run, I really don't want it stopping on a typing issue. Do you think the suggestion is safe enough for this scenario?

Reply:
Since this is a change that affects static typing, you could figure out if `mypy` is upset with this _before running the extraction process_. So, yes, it's safe!

Reply:
Since "word" and "sentence" are values that can come from code (they don't come from any sort of user input), this is safe!

Here are some further suggestions:

```python
# import

from __future__ import annotations
from typings import Literal

# globals
WordOrSentence = Literal["word"] | Literal["sentence"]

# the named tuple:
class Segment(NamedTuple):
    # ...
    type: WordOrSentence

# definition of extract_data
def extract_data(
    _file, _type: WordOrSentence, snippet, audio, speaker, session_id, english_tier, comment_tier
) -> tuple:
```

Reply:
Thanks for the suggestions and clarification!